lib/galaxy/tool_util/verify/interactor.py

```diff
@@ -1516,6 +1513,8 @@ def register_exception(e: Exception):
 
     job_stdio = galaxy_interactor.get_job_stdio(job["id"])
 
+    assert data_list or data_collection_list, "Tool produced no output datasets or collections"
```

Comment:
That's only a bug if we expected an output. Pure export tools don't need to produce outputs.